packages/url-utils/lib/UrlUtils.js

```diff
@@ -40,11 +55,34 @@ module.exports = class UrlUtils {
         this._config.staticFilesUrlPrefix = STATIC_FILES_URL_PREFIX;
         this._config.staticMediaUrlPrefix = STATIC_MEDIA_URL_PREFIX;
 
+        const assetBaseUrls = options.assetBaseUrls || {};
+        this._assetBaseUrls = {
+            image: trimTrailingSlash(assetBaseUrls.image || null),
+            files: trimTrailingSlash(assetBaseUrls.files || null),
+            media: trimTrailingSlash(assetBaseUrls.media || null)
+        };
+
         this.getSubdir = options.getSubdir;
         this.getSiteUrl = options.getSiteUrl;
         this.getAdminUrl = options.getAdminUrl;
     }
 
+    _assetOptionDefaults() {
+        const siteBase = trimTrailingSlash(this.getSiteUrl ? this.getSiteUrl() : null);
```

Comment:
I don't think we need the conditional here - we unconditionally use this method throughout the class